src/alloc_pool.c

```diff
@@ -164,14 +242,17 @@ void *qd_alloc(qd_alloc_type_desc_t *desc, qd_alloc_pool_t **tpool)
         desc->stats->held_by_threads += desc->config->transfer_batch_size;
 #endif
         for (idx = 0; idx < desc->config->transfer_batch_size; idx++) {
-            item = DEQ_HEAD(desc->global_pool->free_list);
-            DEQ_REMOVE_HEAD(desc->global_pool->free_list);
-            DEQ_INSERT_TAIL(pool->free_list, item);
+            item = pop_stack(&desc->global_pool->free_list);
+            push_stack(&pool->free_list, item);
         }
     } else {
         //
         // Allocate a full batch from the heap and put it on the thread list.
         //
+        //TODO(franz):
+        //  -   would be better to allocate in batches == transfer_batch_size
+        //      and put a small (== sizeof(transfer_batch_size)) ref_count to help the final free
+        //  -   could be beneficial directly to delink a chunk?
```

Comment:
I need to explain it better IMHO. Let me rephrase something that I haven't clarified yet on the JIRA.
The behaviour of this chunked stack is not actually the same of the original one, in particular while "rebalancing" resources into/from the global free list.
I have chosen to just move `qd_alloc_item` between free lists, but not `qd_alloc_chunk_t` on purpose, because:

- the purpose of `qd_alloc_chunk_t` is to cope with the allocation rate of `qd_alloc_item` on a same thread
- to reduce the need to move "big" chunks  between threads (causing additional cache misses on the first attempt to access them)
- thread local lists are nearly bounded in size `localFreeListMax` ie can't grow unbounded (maybe)

I'm not sure I've chosen right, that's the reason of the second comment ie 
> could be beneficial directly to delink a chunk?

Maybe! I need a paper and a pencil to think about the implications of it and which benefits will bring!